pkg/apis/nmstate/v1alpha1/nmstate_types.go

```diff
@@ -11,6 +11,17 @@ import (
 type NMstate struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              NMstateSpec `json:"spec,omitempty"`
+}
+
+// NMstateSpec defines the desired state of NMstate
+type NMstateSpec struct {
+	// ExtraNodeSelector is an optional selector that will be added to handler DaemonSet manifest
+	// for both workers and masters (https://github.com/nmstate/kubernetes-nmstate/blob/master/deploy/handler/operator.yaml).
+	// If ExtraNodeSelector is specified, the handler will run only on nodes that have each of the indicated key-value pairs
+	// as labelsbe applied to the node.
+	// +optional
+	ExtraNodeSelector map[string]string `json:"extranodeselector"`
```

Comment:
I think that the extra node is extra. Maybe just NodeSelector? And please make the json to nodeSelector.

Reply:
+1